File: day-3/part-1.js
const { read } = require('../utils')

;(async () => {
  const data = await read('./data.txt')
  
  const right = 3
  const down = 1
  let row = down
  let column = right

  let numTrees = 0
  while (row < data.length) {
    if (data[row][column] === '#') {
      numTrees += 1
    }

    row += down

    const isColumnNearBound = column + right >= data[down].length
    column = isColumnNearBound ? column - data[down].length + right : column + right
  }

  console.log('num trees: ', numTrees)
})()

File: day-3/part-1.py
with open('data.txt') as f:
  data = f.read().splitlines()

  right = 3
  down = 1
  row = down
  column = right

  numTrees = 0

  while row < len(data):
    if data[row][column] == '#':
      numTrees += 1

    row += down

    isColumnNearBound = column + right >= len(data[down])
    column = column - len(data[down]) + right if isColumnNearBound else column + right
  
  print(numTrees)